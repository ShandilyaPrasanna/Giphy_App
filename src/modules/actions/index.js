const apiKey = "hi3KViyvIjSvHnq3kCrP1kW4iv1SHGjP";
const domain = "http://api.giphy.com/v1/gifs";
const GET_GIFIMAGES = "GET_GIFIMAGES";
const IS_LOADING = "IS_LOADING";

export function getGIFSByKeywordAction(keyword, limit) {
  return function(dispatch) {
    fetch(`${domain}/search?q=${keyword}&api_key=${apiKey}&limit=${limit}`, {
      method: "GET",
      headers: { "Content-Type": "application/json" }
    })
      .then(
        response => {
          if (response.status === 200) {
            return response.json();
          }
        },
        error => {
          console.error("Something went wrong", error);
          dispatch(isLoading(false));
        }
      )
      .then(
        res => {
          dispatch({ type: GET_GIFIMAGES, payload: res });
        },
        error => {
          console.error("Something went wrong", error);
          dispatch(isLoading(false));
        }
      );
  };
}

export function isLoading(bool) {
  return { type: IS_LOADING, payload: bool };
}

export function getTrendingGIFS(limit) {
  return function(dispatch) {
    fetch(`${domain}/trending?api_key=${apiKey}&limit=${limit}`, {
      method: "GET",
      headers: { "Content-Type": "application/json" }
    })
      .then(
        response => {
          if (response.status === 200) {
            return response.json();
          }
        },
        error => {
          console.error("Something went wrong", error);
          dispatch(isLoading(false));
        }
      )
      .then(
        res => {
          dispatch({ type: GET_GIFIMAGES, payload: res });
        },
        error => {
          console.error("Something went wrong", error);
          dispatch(isLoading(false));
        }
      );
  };
}
