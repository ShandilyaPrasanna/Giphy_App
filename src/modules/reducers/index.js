import { combineReducers } from "redux";
import gifReducer from "./getGIFReducer";

export const reducers = combineReducers({
  app: gifReducer
});
