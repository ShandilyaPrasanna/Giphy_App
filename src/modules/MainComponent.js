import React, { Component } from "react";
import Grid from "./components/Grid";
import { connect } from "react-redux";
import { getGIFSByKeywordAction, getTrendingGIFS, isLoading } from "./actions";

export class MainComponent extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchText: ""
    };
  }

  componentDidMount() {
    this.props.getTrendingGIFS(25);
    this.props.isLoading(true);
  }

  handleOnChange = event => {
    if (event.key === "Enter") {
      this.props.getGIFSByKeywordAction(event.target.value, 25);
      this.props.isLoading(true);
    }
    this.setState({ searchText: event.target.value });
  };

  displayHeader = () => {
    return (
      <div className="logo">
        <img
          src="https://developers.giphy.com/static/img/dev-logo-lg.7404c00322a8.gif"
          width="200px"
        />
      </div>
    );
  };

  displaySearchBar = () => {
    return (
      <div className="wrap">
        <div className="search">
          <input
            id="seachBox"
            type="text"
            className="searchTerm"
            placeholder="Press Enter To Search...."
            onKeyPress={this.handleOnChange}
          />
        </div>
      </div>
    );
  };

  render() {
    return (
      <div>
        {this.props.loading && <div className="spinner" />}
        {this.displayHeader()}
        {this.displaySearchBar()}
        <Grid searchText={this.state.searchText} />
      </div>
    );
  }
}

export function mapStateToProps(state) {
  return {
    loading: state.app.loading
  };
}

export default connect(
  mapStateToProps,
  {
    getGIFSByKeywordAction,
    getTrendingGIFS,
    isLoading
  }
)(MainComponent);
