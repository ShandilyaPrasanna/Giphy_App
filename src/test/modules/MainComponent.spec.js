import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import toJson from "enzyme-to-json";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { MainComponent, mapStateToProps } from "../../modules/MainComponent";

const state = {
  app: {
    loading: true,
    images: [],
    pagination: {}
  }
};

configure({ adapter: new Adapter() });
describe("MainComponent Test", () => {
  let tree;
  const props = {
    getGIFSByKeywordAction: jest.fn(),
    getTrendingGIFS: jest.fn(),
    isLoading: jest.fn(),
    loading: true
  };
  beforeEach(() => {
    tree = shallow(<MainComponent {...props} />);
  });

  it("renders MainComponent ", () => {
    expect(toJson(tree)).toMatchSnapshot();
  });

  it("check onChange handler ", () => {
    tree.find("#seachBox").simulate("keyPress", { target: { value: "apple" } });
    expect(tree.state("searchText")).toBe("apple");
  });
  it("mapStateToProps", () => {
    expect(mapStateToProps(state).loading).toBe(true);
  });
});
