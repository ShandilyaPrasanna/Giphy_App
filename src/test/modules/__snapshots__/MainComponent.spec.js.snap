// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainComponent Test renders MainComponent  1`] = `
<div>
  <div
    className="spinner"
  />
  <div
    className="logo"
  >
    <img
      src="https://developers.giphy.com/static/img/dev-logo-lg.7404c00322a8.gif"
      width="200px"
    />
  </div>
  <div
    className="wrap"
  >
    <div
      className="search"
    >
      <input
        className="searchTerm"
        id="seachBox"
        onKeyPress={[Function]}
        placeholder="Press Enter To Search...."
        type="text"
      />
    </div>
  </div>
  <ConnectFunction
    searchText=""
  />
</div>
`;
